### ISSUE TITLE
v8: Imagepicker prevalue editor is broken

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco-CMS#4911]
It seems in v8 it is no longer possible to use "imagepicker" prevalue editor in grid settings/styles as in v7.

https://our.umbraco.com/documentation/getting-started/backoffice/property-editors/built-in-property-editors/Grid-Layout/Settings-and-styles#sample-settings

Insert this json in a grid styles in the config.

```
{
    "label": "Background image",
    "description": "Choose an image",
    "key": "background-image",
    "view": "imagepicker",
    "modifier": "url('{0}')"
}
```

![image](https://user-images.githubusercontent.com/2919859/53989689-d2a37080-4126-11e9-8143-f2a595758953.png)

![image](https://user-images.githubusercontent.com/2919859/53989716-e4851380-4126-11e9-9933-8ef51a6472ed.png)

The `imagepicker` prevalue editor exists here https://github.com/umbraco/Umbraco-CMS/blob/dev-v8/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/imagepicker.html

I guess it is because the `umb-overlay` here use `mediapicker` view in `umb-overlay` directive:
https://github.com/umbraco/Umbraco-CMS/blob/91c52cffc8b7c70dc956f6c6610460be2d1adc51/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/imagepicker.controller.js#L4-L19

But in v8 this overlay has been removed.
https://github.com/umbraco/Umbraco-CMS/tree/91c52cffc8b7c70dc956f6c6610460be2d1adc51/src/Umbraco.Web.UI.Client/src/views/common/overlays

For example `itempicker` doesn't return an error. Also see this issue: https://github.com/umbraco/Umbraco-CMS/issues/4909

**Comment by bjarnef** [comment]
We should probably review if the changes in overlays from v7 to v8 breaks other prevalue editors using these views in `umb-overlay` or other parts of the UI reference to overlays that no longer exists.

**Overlays**
https://github.com/umbraco/Umbraco-CMS/tree/dev-v7/src/Umbraco.Web.UI.Client/src/views/common/overlays

https://github.com/umbraco/Umbraco-CMS/tree/dev-v8/src/Umbraco.Web.UI.Client/src/views/common/overlays

**Prevalue editors**
https://github.com/umbraco/Umbraco-CMS/tree/dev-v7/src/Umbraco.Web.UI.Client/src/views/prevalueeditors

https://github.com/umbraco/Umbraco-CMS/tree/dev-v8/src/Umbraco.Web.UI.Client/src/views/prevalueeditors

**Comment by bjarnef** [comment]
It might also be worth to group prevalue editors in folders just like property editors are grouped in folders.

**Comment by bjarnef** [comment]
The `imagepicker` should either use `editorService` to open `mediapicker` infinite editor or there should be added a `mediapicker` overlay.

**Comment by DavidDigitul** [comment]
I am experiencing this same issue, has anyone found a work around?

**Comment by bjarnef** [comment]
@DavidDigitul the `imagepicker` property editor should probabaly use `editorService` instead and use the infinite media picker.

**Comment by nul800sebastiaan** [comment]
I'm in the process of applying the PR to v8.0.2, due out next Tuesday.

**Comment by nul800sebastiaan** [comment]
All done, commit: https://github.com/umbraco/Umbraco-CMS/commit/f82dff597b7e57ac81f7e587cf2fc6132cca680b

**Comment by patriciabobes** [comment]
Hi everyone! Any examples of the workaround? Is not working for me.

**Comment by bjarnef** [comment]
@patriciabobes you can add the following json object to the grid settings or styles config.
It has been fixed in v8.1.0 (not sure if it only has been fixed in v8.0.2)

```
{
    "label": "Background image",
    "description": "Choose an image",
    "key": "background-image",
    "view": "imagepicker",
    "modifier": "url('{0}')"
}
```


Alternative you can use `mediapicker` although it is not ideal since it allow selecting folder and files as well.
```
{
    "label": "Background image",
    "description": "Choose an image",
    "key": "background-image",
    "view": "mediapicker",
    "modifier": "url('{0}')"
}
```

**Comment by bjarnef** [comment]
An option if also to create your own prevalue editor and just copying these two files and put them in a folder e.g. `/App_Plugins/PrevalueEditors/ImagePicker/`:
https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/imagepicker.html
https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/imagepicker.controller.js

You should probably also rename the controller name in the controller and view to something else so it doesn't conflict with the existing one `Umbraco.PrevalueEditors.ImageFilePickerController`.

```
{
    "label": "Background image",
    "description": "Choose an image",
    "key": "background-image",
    "view": "/App_Plugins/PrevalueEditors/ImagePicker/imagepicker.html",
    "modifier": "url('{0}')"
}
```
